src/tools/imgtool/modules/hp85_tape.cpp

```diff
@@ -0,0 +1,1192 @@
+
+    HP-85 tape format
+
+*********************************************************************/
+
+#include "imgtool.h"
+#include "formats/imageutl.h"
+#include "formats/hti_tape.h"
+#include <iostream>
+
+// Constants
+static constexpr unsigned CHARS_PER_FNAME = 6;  // Characters in a filename
+static constexpr unsigned CHARS_PER_EXT = 4;    // Characters in (simulated) file extension
+static constexpr unsigned CHARS_PER_FNAME_EXT = CHARS_PER_FNAME + 1 + CHARS_PER_EXT;    // Characters in filename + extension
+static constexpr unsigned MAX_FILE_NO = 42;     // Maximum file #
+static constexpr unsigned MAX_RECORD_SIZE = 256;    // Maximum size of record body
+static constexpr unsigned MAX_N_RECORDS = 436 * 2 - 5;  // Total user-available capacity of tape (in physical records)
+static constexpr unsigned DIR_RECORDS = 4;  // Records reserved to directory
+
+// Words stored on tape
+using tape_word_t = hti_format_t::tape_word_t;
+
+// Tape position, 1 unit = 1 inch / (968 * 1024)
+using tape_pos_t = hti_format_t::tape_pos_t;
+
+// File number [0..42]
+typedef uint8_t file_no_t;
+
+// Minimum gap size to detect IFGs: 1.25"
+static constexpr tape_pos_t MIN_IFG_SIZE = 1.25 * hti_format_t::ONE_INCH_POS;
+
+// Minimum gap size to detect IRGs: 1/32 "
+static constexpr tape_pos_t MIN_IRG_SIZE = hti_format_t::ONE_INCH_POS / 32;
+
+// Formatted size of IFGs: 2.5"
+static constexpr tape_pos_t FMT_IFG_SIZE = 2.5 * hti_format_t::ONE_INCH_POS;
+
+// Formatted size of IRGs: 1"
+static constexpr tape_pos_t FMT_IRG_SIZE = hti_format_t::ONE_INCH_POS;
+
+// Formatted size of records: 2.85"
+static constexpr tape_pos_t FMT_REC_SIZE = 2.85 * hti_format_t::ONE_INCH_POS;
+
+// Starting position on tracks: 74" from beginning of tape
+static constexpr tape_pos_t TRACK_START = 74 * hti_format_t::ONE_INCH_POS;
+
+// Sync word: 0x0001
+static constexpr tape_word_t SYNC_WORD = 1;
+
+// Name of erased (NULL) files
+static const char *const NULL_FILENAME = "==NULL==";
+
+// Masks of bits in file type
+static constexpr uint8_t FT_NEXT_AV_MASK = 0x80;    // Next available slot
+static constexpr uint8_t FT_NULL_FILE_MASK = 0x40;  // Erased file
+static constexpr uint8_t FT_PROG_MASK = 0x20;       // PROG (BASIC) file
+static constexpr uint8_t FT_DATA_MASK = 0x10;       // DATA file
+static constexpr uint8_t FT_BPGM_MASK = 0x08;       // BPGM file
+static constexpr uint8_t FT_WP_MASK = 0x02;         // Write protection
+static constexpr uint8_t FT_HIDDEN_MASK = 0x01;     // Hidden file
+
+/********************************************************************************
+ * Directory entries
+ ********************************************************************************/
+typedef struct {
+	uint8_t filename[ CHARS_PER_FNAME ];  // Filename (left justified, space-padded on the right)
+	file_no_t file_no;  // File #
+	uint8_t filetype;   // File type
+	uint16_t n_recs;    // Physical records
+	uint16_t record_len;    // Length of logical records
+} dir_entry_85_t;
```

Comment:
Since we're using C++ here, could you please use C++-style declaration of structres?  i.e. `struct name { ... };` declares a structure type in C++, not just a struct tag.  Also, names ending with `_t` are reserved at global scope according to POSIX - I realise MAME already violates this in a few places, but no need to add to it.